sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.rest.PagedFlux;
+import java.util.List;
+import java.util.Map;
+import reactor.core.publisher.Mono;
+
+/**
+ * class for the table async client
+ */
+@ServiceClient(
+    builder = TableClientBuilder.class,
+    isAsync = true)
+public class TableAsyncClient {
+    private final String tableName;
+
+    TableAsyncClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * Queries and returns entities in the given table using the select and filter strings
+     *
+     * @param queryOptions the odata query object
+     * @return a paged flux of all the entity which fit this criteria
+     */
+    public PagedFlux<TableEntity> queryEntity(QueryOptions queryOptions) {
+        return null;
+    }
+
+    /**
+     * Queries and returns entities in the given table with the given rowKey and ParitionKey
+     *
+     * @param rowKey the given row key
+     * @param partitionKey the given partition key
+     * @return a list of the tables that fit the row and partition key
+     */
+    public PagedFlux<TableEntity> queryEntitiesWithPartitionAndRowKey(String rowKey, String partitionKey) {
+        return null;
+    }
+
+    /**
+     * insert a TableEntity with the given properties and return that TableEntity. Property map must include
+     * rowKey and partitionKey
+     *
+     * @param tableEntityProperties a map of properties for the TableEntity
+     * @return the created TableEntity
+     */
+    public Mono<TableEntity> createEntity(Map<String, Object> tableEntityProperties) {
+        return Mono.empty();
+    }
+
+    /**
+     * based on Mode it either inserts or merges if exists or inserts or merges if exists
+     *
+     * @param updateMode type of upsert
+     * @param tableEntity entity to upsert
+     * @return void
+     */
+    public Mono<Void> upsertEntity(UpdateMode updateMode, TableEntity tableEntity) {
+        return Mono.empty();
+    }
+
+    /**
+     * based on Mode it either updates or fails if it does exists or replaces or fails if it does exists
+     *
+     * @param updateMode type of update
+     * @param tableEntity entity to update
+     * @return void
+     */
+    public Mono<Void> updateEntity(UpdateMode updateMode, TableEntity tableEntity) {
+        return Mono.empty();
+    }
```

Comment:
These methods should return `Mono<TableEntity>` - the updated/upserted entity.

Reply:
I brought this up during our tables meeting and we decided we have to talk with the Tables service team since the Merge API call returns at 204 (no content) so we are still deciding how best to return something if that underlying functionality doesn't return an entity. So we are asking them about hero scenarios and what users would want from this type of function